### ISSUE TITLE
`make build` should ensure minimum version of poetry is installed

### ISSUE THREAD

**Issue by foragerr** [OpenDevin/OpenDevin#746]
#### Describe the bug
See: https://github.com/OpenDevin/OpenDevin/issues/740#issuecomment-2038577787
if there is an older version of poetry already installed, the [installation step in Makefile](https://github.com/OpenDevin/OpenDevin/blob/main/Makefile#L19) does not upgrade poetry.

if installed poetry is old enough, it wont recognize the `--without` option and fail.

**Comment by dorbanianas** [comment]
I tried to reproduce this with many versions but it seems that it's working.

poetry initial versions: 1.1.12, 1.2.0

**Comment by dorbanianas** [comment]
I don't know if you have more details about how we can reproduce it

**Comment by foragerr** [comment]
I based this issue on the other user's experience, I haven't tried to reproduce it myself. Will try in a bit.

**Comment by dorbanianas** [comment]
Yeah thank you for your help

**Comment by zhonggegege** [comment]
`  NoSuchOptionException

  The "--without" option does not exist.

  at /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:300 in _add_long_option
      296│     def _add_long_option(
      297│         self, name, value, tokens, fmt, lenient
      298│     ):  # type: (str, Optional[str], List[str], ArgsFormat, bool) -> None
      299│         if not fmt.has_option(name):
    → 300│             raise NoSuchOptionException(name)
      301│ 
      302│         option = fmt.get_option(name)
      303│ 
      304│         if value is False:
make: *** [Makefile:20: build] Error 1`


Test poetry 1.1.12 version does not recognize "--without",i'm use this

**Comment by foragerr** [comment]
@zhonggegege did you get this when you ran `make build` ?

**Comment by foragerr** [comment]
I'm unable to repro either. Closing this issue.

Clean install of poetry 1.1.12:
```
# docker run --rm -it python:3.12 /bin/bash
# apt update && apt install -y pipx
# pipx install poetry==1.1.12

# poetry --version
Poetry version 1.1.12
```

Attempt installation using the command from Makefile
```
# curl -sSL https://install.python-poetry.org | python3 -
Installing Poetry (1.8.2): Done

# poetry --version
Poetry (version 1.8.2)
```

**Comment by zhonggegege** [comment]
> @zhonggegege你跑步时得到这个吗`make build`？



> make build

Thank you for your reply. I'm sorry that it took me so long to reply to you. After testing, I found that it was a conflict between Poetry versions in the global environment. I uninstalled Poetry under the system and reinstalled Poetry 1.8.2 in an independent conda environment. After that, the problem was solved and everything works normally. `make build` done.
